expression/builtin_encryption.go

```
@@ -625,7 +626,13 @@ func (b *builtinMD5Sig) evalString(row chunk.Row) (string, bool, error) {
 	if isNull || err != nil {
 		return "", isNull, err
 	}
-	sum := md5.Sum([]byte(arg)) // #nosec G401
+	var sum [16]byte
+	dBytes, err := charset.NewEncoding(b.args[0].GetType().Charset).Encode(nil, []byte(arg))
+	if err == nil {
+		sum = md5.Sum(dBytes) // #nosec G401
+	} else {
```

Do we have a test for this logic?

---

No, we will not go to this branch in TiDB normal use, because of #28778. If it's necessary, I will add it.

---

And a question here, need I add a warning log when get invalid character here?

---

@hawkingrei https://github.com/pingcap/tidb/issues/29314 will handle this problem uniformly? Maybe we can return an error or a panic？

---

@zimulala Updated PTAL

---

> @hawkingrei #29314 will handle this problem uniformly? Maybe we can return an error or a panic？

OK, I will.